### PR TITLE
Initial port of code analysis docs

### PR DESCRIPTION
**Pull request by gewarren**
- Contributes to #20211.
- [Preview link](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/productivity/code-analysis?branch=pr-en-us-20213).
- Add new [msbuild props](https://review.docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props?branch=pr-en-us-20213#code-analysis-properties).
- Create a new parent folder: docs/fundamentals/ and docs/fundamentals/productivity.
- For now, I'm putting the new articles in the docs/core TOC, but they will move in the future.

### CODE REVIEW

File: docs/fundamentals/productivity/code-analysis.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+helpviewer_keywords:
+- code analysis
+- code analyzers
+---
+# Overview of .NET source code analysis
+
+.NET code analyzers inspect your C# or Visual Basic code for security, performance, design, and other issues. Starting in .NET 5.0, these analyzers are included with the .NET SDK. You can also install third party analyzers, such as [StyleCop](https://www.nuget.org/packages/StyleCop.Analyzers/), [Roslynator](https://www.nuget.org/packages/Roslynator.Analyzers/), [XUnit Analyzers](https://www.nuget.org/packages/xunit.analyzers/), and [Sonar Analyzer](https://www.nuget.org/packages/SonarAnalyzer.CSharp/).
+
+> [!TIP]
+> In previous versions of .NET, you can install the .NET code analyzers as a [NuGet package](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers). For more information, see [Install code analyzers](/visualstudio/code-quality/install-fxcop-analyzers).
```
Comment (mavasani) [inline docs/fundamentals/productivity/code-analysis.md:16]: Actually users can enable the .NET analyzers that ship in the box even on earlier projects by setting certain MSBuild property. The only requirement is users need to be on .NET5 SDK or later. @mikadumont is planning to write up a doc for this part. I’ll also forward you an email for your reference.

**Review by mavasani — APPROVED** [review]

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
I think you need to update what's new bot config file as well.

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

I generally agree with @Youssef1313 's suggestions. I'll let you decide which make sense, and which aren't needed.

**Review by mavasani — APPROVED** [review]
Looks great to me. Thanks!

**Pull request merged**